src/modules/plus/filter_strobe.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * filter_strobe.c -- simple strobing filter
+ * Copyright (C) 2020 Martin Sandsmark <martin.sandsmark@kde.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt_filter.h>
+#include <framework/mlt_frame.h>
+#include <framework/mlt_producer.h>
+#include <framework/mlt_service.h>
+#include <framework/mlt_factory.h>
+#include <framework/mlt_property.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+
+static int filter_get_image( mlt_frame frame, uint8_t **image, mlt_image_format *format, int *width, int *height, int writable )
+{
+	// Get the image
+	mlt_filter filter = mlt_frame_pop_service( frame );
+	mlt_properties properties = MLT_FILTER_PROPERTIES( filter );
+
+	int error = mlt_frame_get_image( frame, image, format, width, height, 1 );
+	if ( error )
+	{
+		return error;
+	}
+
+	int invert = mlt_properties_get_int( properties, "strobe_invert" );
+	int interval = mlt_properties_get_int( properties, "interval" );
+	mlt_position currentpos = mlt_filter_get_position( filter, frame );
+
+	int do_strobe = ( currentpos % ( interval + 1 ) ) > interval / 2;
+
+	if ( invert )
+	{
+		do_strobe = !do_strobe;
+	}
+
+	if ( do_strobe != 1 )
+	{
+		return 0;
+	}
+
+	assert( *width >= 0 );
+	assert( *height >= 0 );
+	size_t alphasize = *width * *height;
+	uint8_t *alpha_buffer = mlt_frame_get_alpha_mask( frame );
+	memset( alpha_buffer, 0, alphasize );
```

Comment:
I think you could put this in an else statement below since rgb24a will not need an alpha mask.

Also good to check the pointer for null before accessing it (although failure is unlikely).

Reply:
> I think you could put this in an else statement below since rgb24a will not need an alpha mask.

I'm not sure if that's true for all filters? I. e. if some filters modify the image we still want to completely blank the frame.

I plan to go through the other filters and fix the issue with them ignoring the alpha mask eventually, though, since it's not intuitive that this filter is overridden if it's not last in the chain.


> Also good to check the pointer for null before accessing it (although failure is unlikely).

the API contract (or whatever, the comments at least) states that mlt_frame_get_alpha_mask() will allocate if missing, but I can add a check anyways.

Reply:
>> I think you could put this in an else statement below since rgb24a will not need an alpha mask.
> I'm not sure if that's true for all filters? I. e. if some filters modify the image we still want to completely blank the frame.

Image conversion in MLT takes care of merging and extracting the alpha channel when converting to and from rgba. It is not the intention that alpha channel needs to be redundant on every frame. I am sorry I gave that impression. If a filter is operating in rgba, it can assume the embedded alpha channel is fine. Otherwise, it can use `mlt_frame_get_alpha()` or `mlt_frame_get_alpha_mask()`. Brian is correct that if else can work here.

> I plan to go through the other filters and fix the issue with them ignoring the alpha mask eventually,

That is not desired. I think the problem you are seeing is that some services accidentally overwrite the alpha channel instead of passing it through. Some frei0r filters have an alpha operation parameter, which is nice.

Reply:
One thing that makes alpha channel handling messy in MLT is the `get_alpha_mask` function pointer in `mlt_frame` as used inside `mlt_frame_get_alpha_mask()`. I would like to get rid of that in v7. There is only one thing that uses it (`transition_region.c`), and it can be interesting, but it makes things complicated: if a filter is operating in rgba but the frame has this function pointer set, what should it do? Many things (e.g. frei0r, Qt) cannot handle two sources of alpha channel.